### DISCUSSION TITLE
ParserError first time running plotman

### DISCUSSION THREAD

**Discussion by hutchwilco** [ericaltendorf/plotman#932] [Q&A] [answered]
I've got the following error when running `plotman interactive`
I've checked the yaml, several online yaml validators confirm my check.
The only thing is I'm running on a machine that had chia installed via the experimental .deb file rather than "git install" that I'm used to. So I've not activated an environment when running this. Plotman is installed in /home/<user>/.local/bin/

Any help would be appreciated.

```python-traceback
    raise ParserError("while parsing a block mapping", self.marks[-1],
yaml.parser.ParserError: while parsing a block mapping
  in "<unicode string>", line 44, column 9:
            tmp: /mnt/ram
            ^
expected <block end>, but found '<block mapping start>'
  in "<unicode string>", line 51, column 10:
             tmp2: /mnt/nvme1
             ^
```

**Comment by altendky** [comment, marked as answer]
`tmp:` is expected to be a list, not a string, even if you have only one.

https://github.com/ericaltendorf/plotman/blob/1b7dfba139ed03d3c32a2f04f6011e03bfb1f442/src/plotman/resources/plotman.yaml#L44-L48

I wonder if you also have some inconsistent indentation.

**Reply by hutchwilco** [reply]
Thank you @altendky. I had considered this list issue, but took my cue from the default tmp2: parameter:

https://github.com/ericaltendorf/plotman/blob/1b7dfba139ed03d3c32a2f04f6011e03bfb1f442/src/plotman/resources/plotman.yaml#L52-L53

I'm unsure if this qualifies as a bug, but where I'm from we call this a "bum steer" - misleading of innocent ignorance :)
I've corrected the list vs string issue, which resolved my ParserError.

But then had a "config file malformed" error  - which as you suggested was indeed inconsistent indentation. 
I had followed the "two spaces" indent rule (and yaml checkers clear it as ok) but this is clearly not the plotman way. 
To fix it I copied the indentation exactly from the default plotman.yaml.

[https://github.com/ericaltendorf/plotman/blob/1b7dfba139ed03d3c32a2f04f6011e03bfb1f442/src/plotman/resources/plotman.yaml](url)

Thanks again

**Reply by altendky** [reply]
`tmp2` is a single string, `tmp` is a list of strings.  I don't think the reference is incorrect, at least here, but yes, one of the many things on the todo list is to provide something more than just the example file and wiki for config reference.

As for indentation, if you look at the original error here, `tmp` was at column 9 and `tmp2` was at column 10, they just weren't indented to the same level.  plotman uses [`PyYAML`](https://pypi.org/project/PyYAML/) for the yaml parsing.  I would fully expect that consistent two space indentation would be fine.  One item at a given level being at 9 and the other at 10 shouldn't be valid regardless I would expect.  `:]`

Glad you got through all this.

**Reply by hutchwilco** [reply]
Thanks heaps. 
To clarify, I got the parser error with tmp2: as a single string as the default file has - so I do think this also has to be a list/is a bug.

This caused me to change the tmp: to a single string while attempting to fix things which introduced errors on both tmp: and tmp2: (in retrospect this should have been a clear sign I'd gone in the wrong direction).

**Reply by altendky** [reply]
You can see the definition used to process what comes out of PyYAML here.

https://github.com/ericaltendorf/plotman/blob/1b7dfba139ed03d3c32a2f04f6011e03bfb1f442/src/plotman/configuration.py#L356-L358

If `tmp2` fails with a single string and works with a list that would be good for me to know.  But, the original error is entirely about the misindentation.  I just read it and also knew that the single string for tmp was also wrong.
